proto/buf/alpha/registry/v1alpha1/schema.proto

```diff
@@ -0,0 +1,196 @@
+  // is unknown, the request will fail and no schema will be returned.
+  //
+  // The resulting schema contains these elements and all of their dependencies,
+  // which is enough information for the caller to construct a dynamic message for
+  // any requested types (e.g. requested message types or request and response types
+  // for requested service and method names).
+  repeated string element_names = 5;
+
+  // If true, well-known imports will be included the returned set of files. If
+  // false or not present, these files will omitted from the response (e.g. the
+  // caller is assumed to already know the types they define).
+  //
+  // These files are a subset of the files in the "well-known types" module:
+  // https://buf.build/beta/wellknowntypes
+  //
+  // Specifically, the files that are omitted by default are the following:
+  //  * google/protobuf/any.proto
+  //  * google/protobuf/descriptor.proto
+  //  * google/protobuf/duration.proto
+  //  * google/protobuf/empty.proto
+  //  * google/protobuf/field_mask.proto
+  //  * google/protobuf/struct.proto
+  //  * google/protobuf/timestamp.proto
+  //  * google/protobuf/wrapper.proto
+  //
+  // These include all well-known types that have custom JSON formats and
+  // the types in "descriptor.proto", required for Protobuf reflection.
+  // Callers typically use compiled-in versions of these files.
+  bool include_well_known_imports = 6;
+
+  // If true, the returned schema will be expanded to include custom option
+  // definitions for options present on the requested schema elements.
+  bool include_custom_options = 7;
+
+  // If true, the returned schema will include all known extensions for any
+  // extendable messages among the requested schema elements.
+  bool include_extensions = 8;
+}
+
+message GetSchemaResponse {
+  // The resolved version of the schema. If the request included a reference to a commit,
+  // this value is the same as that commit. If the request included a reference to a tag,
+  // this is the commit version for that tag. If the request did not include any version,
+  // this is the latest version of the requested module.
+  //
+  // This version can be used in subsequent invocations in the only_if_not_version
+  // request field to poll for newer versions of the schema.
+  string version = 1;
+  // The schema, which is a set of file descriptors that include the requested elements
+  // and their dependencies.
+  //
+  // If the request included an only_if_not_version value, and the resolved version
+  // matched that version, this will be absent, indicating that the resolved version
+  // has not changed.
+  google.protobuf.FileDescriptorSet schema_files = 2;
+}
+
+message ReformatRequest {
+  // The module which contains the schema of interest. Required.
+  Module module = 1;
+  // The message type of the input data. Required.
+  string message_name = 2;
+
+  // The input data that is to be reformatted. Required.
+  bytes input_data = 3;
```

Comment:
the attributes in this message become troublesome when we add a noun to the message name. The input bytes should match the chosen noun. ie `input_message`

Reply:
To me `input_message` and `message_name` seem ambiguously named. The `message_name` is what selects the schema for processing the input data.

The other convert endpoint uses the word "payload". Is that better? Maybe `input_payload` (and `output_payload` in the response)?

Reply:
hmm i see what you mean...
```diff
-  bytes input_data = 3;
+  bytes message_data = 3;
```
 which then falls into the next attribute 
```diff
-  Format input_format = 4;
+  Format message_format = 4;
```
so if it looked like 
```proto
message ConvertMessageRequest {
  Module module = 1;
  string name = 2;
  bytes data = 3;
  Format format = 4;
...
}
```
do we hate it?

Reply:
I think `name`, without any qualification`, is a bit vague. I prefer the mild redundancy of having "message" in the method name but also having it in the field name. I prefer keeping `input_` prefixes for fields that define the input, mainly to distinguish them and provide symmetry with analogous elements related to the output. There is, for example, also an `output_format` (in the request, though it's a oneof instead of a field) and an `output_data` (in the response).

Do you think the names as currently shown in the PR are unclear? Or are they inconsistent with naming of these same concepts in other places (other APIs, or docs, etc)?